[IzN432] iP

IzN432 opened the pull request:
# Botty the bot
> "Your mind is for having ideas, not holding them" -David Allen ([source](https://dansilvestre.com/productivity-quotes/))
- Command line style inputs
- **Super** fast and easy to use
- Records your tasks for you

All you need to do is
1. Download the jar file
2. Run it! `java -jar botty.jar`
3. Let *Botty* handle all your tasks :shipit:

Here is the main method:
```java
public static void main(String[] args) {
    Botty botty = new Botty();

    botty.run();
}
```
Here is a sample task list:
- [x] do homework
- [ ] attend lecture
- [ ] attend tutorial

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

IzN432 committed:
Rename Duke to Botty

IzN432 committed:
Add greeting and goodbye messages

IzN432 committed:
Update greeting and exit messages

IzN432 committed:
Add echo feature

IzN432 committed:
Add list functionality

IzN432 committed:
Add mark and unmark functionalities

IzN432 committed:
Move Task class to its own file

IzN432 committed:
Add Event Deadline and Todo classes

IzN432 committed:
Fix style issue

IzN432 committed:
Add ability to add Todos, Events and Deadlines

IzN432 committed:
Add handling for empty list

IzN432 committed:
Change message for unfamiliar command

IzN432 committed:
Rename addToTasklist to addToTaskList

IzN432 committed:
Replace emoticon with Botty

IzN432 committed:
Add text UI testing

IzN432 committed:
Abstract out printing of Botty replies

IzN432 committed:
Standardise indentation of Botty replies

IzN432 committed:
Add handling for user input

IzN432 committed:
Add handling for empty list in setTaskCompletion

IzN432 committed:
Standardise list empty message

IzN432 committed:
Standardise Todo creation

IzN432 committed:
Remove unused import

IzN432 committed:
Abstract handling of each command into functions

IzN432 committed:
Add acknowledgement for isNumber function

IzN432 committed:
Add handling for empty description todo

IzN432 committed:
Add BottyException for error handling

IzN432 committed:
Add extra cases for testing

IzN432 committed:
Change taskList to ArrayList

IzN432 committed:
Add delete feature

IzN432 committed:
Add check for null input

IzN432 committed:
Fix incorrect error message in handleDelete

IzN432 committed:
Add more robust testing including delete feature

IzN432 committed:
Change reply function to accept string instead

IzN432 committed:
Move all task logic to TaskManager class

IzN432 committed:
Remove unused method in Task class

IzN432 committed:
Change Botty's functions into instance methods

IzN432 committed:
Change taskManager from class to instance variable

IzN432 committed:
Refactor argument parsing into ParsedInput class

IzN432 committed:
Refactor commands into individual classes

IzN432 committed:
Package files for better visibility

IzN432 committed:
Remove unnecessary initialization

IzN432 committed:
Set TaskManager fields to final

IzN432 committed:
Shift ParseInput into commands package

IzN432 committed:
Shift classes around

IzN432 committed:
Add storage function

IzN432 committed:
Merge branch 'branch-Level-7'

IzN432 committed:
Refactor task manager to be field in Botty

IzN432 committed:
Add LocalDate handling for all dates

IzN432 committed:
Refactor UI related functions into UI class

IzN432 committed:
Merge branch 'branch-A-MoreOOP'

IzN432 committed:
Refactor commands classes

IzN432 committed:
Packaged everything into one package named botty

IzN432 committed:
Edit runtest.bat to accomodate previous change

IzN432 committed:
Merge branch 'add-gradle-support'

IzN432 committed:
Add gradle support

IzN432 committed:
Fix style check issues

IzN432 committed:
Add JUnit tests

IzN432 committed:
Add JavaDocs to all classes

IzN432 committed:
Make tasks save on every edit

IzN432 committed:
Merge branch 'A-JavaDoc'

IzN432 committed:
Add JavaDocs for isExit and changesTaskList

IzN432 committed:
Add find feature

IzN432 committed:
Adjust taskmanager test case

IzN432 committed:
Fix style errors

sooyitao commented on src/main/java/botty/tasks/TaskManager.java:
```diff
@@ -0,0 +1,145 @@
+     * @throws TaskListEmptyException if the task list is empty
+     * @throws TaskNumberNotFoundException if the task number is out of range
+     */
+    public Task markTask(int index) throws TaskListEmptyException, TaskNumberNotFoundException {
```
Maybe `index` can be renamed to `taskIndex` to make it clear that it’s the index of a task in the list?

sooyitao commented on src/main/java/botty/ui/UI.java:
```diff
@@ -0,0 +1,68 @@
+     * Prints the given content
+     * @param content the content to be printed
+     */
+    public void reply(String content) {
```
The method name `reply` can be descriptive. Should it be `displayReply` instead as it displays a reply to the user and is consistent with `displayIntroduction`?

sooyitao commented on src/main/java/botty/tasks/Task.java:
```diff
@@ -0,0 +1,79 @@
+ */
+public abstract class Task {
+    // Whether the task is completed
+    private boolean completed;
```
Perhaps renaming `completed` to `isCompleted` to improve clarity and consistency, particularly for boolean fields, where the prefix `is` is often used.

IzN432 replied:
will do!

sooyitao reviewed (commented):
LGTM. Overall the code was easy to follow and mostly follows the code quality guidelines. Well done!

IzN432 committed:
Rename complete to isComplete

IsaacPangTH commented on src/main/java/botty/tasks/TaskManager.java:
```diff
@@ -0,0 +1,145 @@
+ */
+public class TaskManager {
+    // The list of tasks
+    private final ArrayList<Task> taskList = new ArrayList<>(100);
```
Maybe rename taskList to tasks to follow the coding standard more closely

IsaacPangTH commented on src/main/java/botty/tasks/TaskManager.java:
```diff
@@ -0,0 +1,145 @@
+    /**
+     * Returns the current number of tasks in the task list
+     */
+    public int size() {
```
Could this method be renamed as a verb?

IsaacPangTH reviewed (commented):
LGTM. Just some nits regarding the Java coding standard.

IzN432 committed:
Add JavaFX UI

IzN432 committed:
Merge branch 'branch-Level-10'

IzN432 committed:
Fix style check errors

IzN432 committed:
Fix style issue

IzN432 committed:
Merge branch 'branch-A-CodingStandard'

IzN432 committed:
Name regex string for clarity

IzN432 committed:
Refactor empty description check to be in Task

IzN432 committed:
Add update functionality

IzN432 committed:
Rewrite find implementation using stream

IzN432 committed:
Merge branch 'branch-A-Stream'

IzN432 committed:
Merge branch 'branch-BCD-Extension'

IzN432 committed:
Add assertion to ensure task not null

IzN432 committed:
Extract regex pattern into named variable

Inline regex string for task data string validation is currently
hardcoded within the condition.

This makes the code harder to understand.

Let's extract the regex pattern into a named variable
`standardDataStorageFormatRegex` to improve code readability.

This approach allows for a more maintainable codebase.

IzN432 committed:
Refactor to fix SLAP and code length violations

Command initialization in `Botty` constructor is mixed with higher
level methods, violating SLAP. `ParsedInput.parse` method is
longer than 30 lines long.

This leads to less maintainable code as lower level details are
included in the `Botty` constructor and the `ParsedInput.parse`
method is too long.

Let's refactor lower level details in both methods into separate
methods.

This improves code readibility to follow SLAP and code length limit.

IzN432 committed:
Rename variables and methods to be clearer

IzN432 committed:
Replace raw type usage with bounded wildcard

Some Command classes include raw type usage of Task.

This causes a raw type usage warning by the compiler.

Let's replace raw type uses with bounded wildcard to eliminate the
error.

This eliminates the raw type usage warning.

IzN432 committed:
Merge pull request #5 from IzN432/branch-A-CodeQuality

Fix code quality issues

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-fix-raw-type-usage

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-A-Assertions

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-change-names

IzN432 committed:
Merge pull request #2 from IzN432/branch-A-Assertions

Add assertions

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-change-names

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-fix-raw-type-usage

IzN432 committed:
Merge pull request #6 from IzN432/branch-fix-raw-type-usage

Replace raw type usage with bounded wildcard

IzN432 committed:
Revert "Replace raw type usage with bounded wildcard"

IzN432 committed:
Merge pull request #7 from IzN432/revert-6-branch-fix-raw-type-usage

Revert "Replace raw type usage with bounded wildcard"

IzN432 committed:
Merge pull request #8 from IzN432/branch-fix-raw-type-usage

Replace raw type usage with bounded wildcard

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-change-names

IzN432 committed:
Merge pull request #4 from IzN432/branch-change-names

Change variable and method names to be clearer

IzN432 committed:
Add user guide

IzN432 committed:
Fix incorrect regex for data string

IzN432 committed:
Add title to stage

IzN432 committed:
Merge pull request #11 from IzN432/branch-A-UserGuide

Add a user guide

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-fix-data-issue

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-add-title-to-stage

IzN432 committed:
Merge pull request #10 from IzN432/branch-fix-data-issue

Fix an issue with the regex processing the data strings

IzN432 committed:
Merge branch 'master' of https://github.com/IzN432/ip into branch-add-title-to-stage

IzN432 committed:
Merge pull request #9 from IzN432/branch-add-title-to-stage

Add title to stage

IzN432 committed:
Enable assertions

IzN432 committed:
Merge pull request #12 from IzN432/branch-enable-assertion

Enable assertion

IzN432 committed:
Refactor nested try catch in `loadTaskList`

IzN432 committed:
Merge pull request #13 from IzN432/branch-simplify-loadTaskList

Refactor nested try catch

IzN432 committed:
Make bye command close the application

IzN432 committed:
Update README.md to match new bye command

IzN432 committed:
Merge pull request #14 from IzN432/branch-bye

Update bye command to exit the application

IzN432 committed:
Add more test cases for TaskManager

IzN432 committed:
Merge pull request #15 from IzN432/branch-more-test-cases

Add more test cases for TaskManager

IzN432 committed:
Fix checkstyle errors in StorageHandler.java

IzN432 committed:
Merge pull request #16 from IzN432/branch-fix-checkstyle-issues

Fix checkstyle errors in StorageHandler.java
